study/src/org/labkey/study/model/DatasetDomainKind.java

```
@@ -435,6 +443,231 @@ else if (categoryId != null)
+
+        try
+        {
+            List<String> allErrors = DomainUtil.updateDomainDescriptor(original, update, container, user).getAllErrors();
+            for (String str : allErrors) {
+                exception.addGlobalError(str);
+            }
+            return exception;
+        }
+        finally
+        {
+            StudyManager.getInstance().uncache(def);
+        }
+    }
+
+    private ValidationException updateDataset(DatasetDomainKindProperties datasetProperties, String domainURI, ValidationException exception,
+                                              StudyImpl study, Container container, User user, DatasetDefinition def)
+    {
+        try
+        {
+            // Update-case specific validation
+            if ( datasetProperties.isDemographicData() && !datasetProperties.isDemographicData() && !StudyManager.getInstance().isDataUniquePerParticipant(def))
+            {
+                String noun = StudyService.get().getSubjectNounSingular(container);
+                throw new IllegalArgumentException("This dataset currently contains more than one row of data per " + noun +
+                        ". Demographic data includes one row of data per " + noun + ".");
+            }
+
+            // PR Note: is it desired that we check for this at all?
+//            if (null != datasetProperties.getCategory() && null == ViewCategoryManager.getInstance().getCategory(container, datasetProperties.getCategory()))
+//                throw new IllegalArgumentException("Unable to find a category named : " + datasetProperties.getCategory() + " in this folder.");
+
+            if (datasetProperties.isDemographicData())
+            {
+                datasetProperties.setKeyPropertyName(null);
+                datasetProperties.setKeyPropertyManaged(false);
+            }
+
+            /* IGNORE illegal shareDataset values */
+            if (!study.getShareVisitDefinitions())
+                datasetProperties.setDataSharing("NONE");

```

The comment here says to ignore the illegal argument for sharedDataset, I think we should instead throw an exception for this case instead of just changing it for the user/caller. I suggest moving this to the `validateDatasetProperties` method and have the check throw an error if the study does not allow for shared visit definitions and the dataSharing property is anything but NONE.